### PR TITLE
Keep generic @extends comments to make post processing easier.

### PR DESCRIPTION
**Pull request by stringham**
This may help with implementing https://github.com/angular/clutz/issues/483, and in the meantime it helps you identify places that need to be manually fixed.

### CODE REVIEW

**Comment by stringham** [comment]
I signed it!

**Comment by rkirov** [comment]
Design-wise, this will make tsickle reject the output due to:

https://github.com/angular/tsickle/blob/master/src/jsdoc.ts#L100

But maybe it is desirable because then the user will go and manually fix the type. @evmar WDYT?

Process-wise, please add a test for this in comments.js / .ts.

**Comment by evmar** [comment]
(I don't have much context, but:)  If you can translate @extends to TS extends then it's better to translate, but if you can't, then probably better to preserve the comment.

**Comment by stringham** [comment]
The idea here is based on #483 where @rkirov said:

>Most likely this is because closure's AST does not support this, but maybe some post processing hack can be applied (at least in this simple case).

It's already creating code that won't compile properly unless it's fixed, and this will help us identify the places that need `<T>` generic braces added.

I agree that it would be ideal to translate the @extends, and clutz seems to be able to gather that information. However clutz does this by getting all of the symbols, and I'm not sure we have access to the symbols when generating typescript in gents.

For some reason, the type information is available when declaring a member type:

```typescript
class MyClass {
    private avlTree: AvlTree<number>;
}
```

comes through okay.

File: src/main/java/com/google/javascript/gents/CommentLinkingPass.java [hunk 1]
```diff
@@ -33,8 +33,11 @@
    */
   private static final Pattern[] JSDOC_REPLACEMENTS = {
     Pattern.compile(
-        BEGIN_JSDOC_LINE + "@(extends|implements|type)[ \t]*(\\{[^@]*\\})[ \t]*(?<keep>)" + EOL,
+        BEGIN_JSDOC_LINE + "@(implements|type)[ \t]*(\\{[^@]*\\})[ \t]*(?<keep>)" + EOL,
         Pattern.DOTALL),
+    //keep generic extends for post processing
```
Comment (rkirov) [inline src/main/java/com/google/javascript/gents/CommentLinkingPass.java]: add a space after '//' and '// TODO: investigate whether generics params can be emitted correctly'.

**Comment by rkirov** [comment]
Yep, I am convinced since we already do the wrong thing, it would be better to keep enough information in the output .ts so that once can at least manually get to the correct state without having to go back to the original .js.

**Comment by rkirov** [comment]
This PR is too stale at this point. Thanks for contributing back, we are still doing many migrations in google and will reopen if this appears again.

**Pull request closed without merging**